compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -4055,11 +4055,17 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
         case _: SingletonType | _: FunOrPolyProto => false
         case _ => true
       var resMatch: Boolean = false
+      // println(i"adaptNoArgs: tree = ${tree}")
+      // println(i"adaptNoArgs: ctx.mode = ${ctx.mode}")
+      // println(i"adaptNoArgs: ptNorm = ${ptNorm}")
+      // println(i"functionExpected = ${functionExpected}")
+      // println(i"needsEta = ${needsEta}")
       wtp match {
         case wtp: ExprType =>
           readaptSimplified(tree.withType(wtp.resultType))
         case wtp: MethodType if wtp.isImplicitMethod &&
-          ({ resMatch = constrainResult(tree.symbol, wtp, sharpenedPt); resMatch } || !functionExpected) =>
+          ({ resMatch = constrainResult(tree.symbol, wtp, sharpenedPt); resMatch } || !functionExpected) &&
+          ((resMatch && !ctx.mode.is(Mode.QuotedPattern)) || !resMatch) =>
```

Comment:
I don't think this is far from clean code, but it's working, at least. We want to figure out how to refactor this.

Reply:
can be simplified to `!(resmatch && ctx.mode.is(Mode.QuotedPattern))`